src/app/campaign-list/campaign-list.service.ts

```diff
@@ -37,34 +55,37 @@ export interface Campaign {
   repName: string;
   notes: string;
   flights?: Flight[];
+  loading?: boolean;
 }
 
 @Injectable()
 export class CampaignListService {
-  params = {page: 1, per: 12};
-  facets: {
-    advertiser: Facet[],
-    podcast: Facet[],
-    status: Facet[],
-    type: Facet[]
-  };
+  params: CampaignParams = {page: 1, per: 12};
+  facets: Facets;
   total: number;
   count: number;
   error: Error;
-  loading = false;
-  flightsLoaded: number;
   // tslint:disable-next-line: variable-name
   private _campaigns = new BehaviorSubject<{[id: number]: Campaign}>({});
+  // tslint:disable-next-line: variable-name
+  private _currentCampaignIds: number[]; // campaign ids for current request/filter params
 
   constructor(private augury: AuguryService) {}
 
+  get loading(): Observable<boolean[]> {
+    return this.campaigns.pipe(
+      map(campaigns => campaigns && campaigns.filter(c => c.loading).map(c => c.loading))
+    );
+  }
+
   get campaigns(): Observable<Campaign[]> {
     // this._campaigns is a subject of campaign entities (an object with campaignIds mapped to campaign,) convert to array for display
     return this._campaigns.asObservable().pipe(
       map((campaignEntities: {}) => {
-        if (Object.keys(campaignEntities).length) {
-          return Object.keys(campaignEntities)
-            // map to array
+        if (this._currentCampaignIds && this._currentCampaignIds.length) {
```

Comment:
this._currentCampaignIds is an array of campaign Ids that were last requested. Since this._campaigns holds all campaigns that have been requested (during the session), I'm limiting what is returned here for display to the current results.